pkg/controller/releasemanager/incarnation.go

```diff
@@ -592,6 +647,31 @@ func (i *IncarnationCollection) unreleasable() []Incarnation {
 	return unreleasable
 }
 
+// alertable returns the incarnations which could currently
+// have alerting enabled
+func (i *IncarnationCollection) alertable() []Incarnation {
+	r := []Incarnation{}
+	for _, i := range i.revisioned() {
+		switch i.status.State.Current {
+		case "canarying":
+			r = append(r, i)
+		case "canaried":
+			r = append(r, i)
+		case "pendingrelease":
+			r = append(r, i)
+		}
+	}
+	for _, i := range i.revisioned() {
```

Comment:
why is this the same loop as above?

Reply:
To put the `canarying|canaried|pendingrelease` states on top, if they exist. The syncer creates the latest `RevisionTarget.AlertRules` based on the states returned by `alertable()`.

 I added a TODO for myself to deprecate this when we deprecate `RevisionTarget.AlertRules`.